spec.bs

```diff
@@ -605,25 +605,31 @@ For now, see [https://github.com/wicg/observable#operators](https://github.com/w
 <div algorithm>
   The <dfn for=Observable method><code>toArray(|options|)</code></dfn> method steps are:
 
+    1. If [=this=]'s [=relevant global object=] is a {{Window}} object, and its [=associated
+       Document=] is not [=Document/fully active=], then return.
```

Comment:
This isn't valid; you need to return a promise.

Also, I don't think you should have this kind of condition? There's no dependency on the document being fully active here.

Reply:
> you need to return a promise.

Oops, yes!

> Also, I don't think you should have this kind of condition? There's no dependency on the document being fully active in the following steps.

Well, I thought we shouldn't be explicitly trying to resolve / reject promises if the document isn't active, right? And that's what the following steps are set up to do. Or did we land on resolving/rejecting being a no-op, so we can just do it without gating on this kind of condition.

Reply:
I generally land on it being a no-op. Like, nothing prevents web developers from doing

```js
const iframeWindow = iframe.contentWindow;
iframe.remove();
const { promise, resolve } = iframeWindow.Promise.withResolvers();
resolve("foo");
```

so especially for "foundational" things like streams or observables, I feel like we should not add extra guards.

Reply:
OK sounds good, I've removed the condition. Thanks.